shared/common-adapters/error-boundary.js

```diff
@@ -42,43 +61,52 @@ const Fallback = ({error, info}: *) => {
             keybase log send
           </Text>
         </Box>}
-      <Text type="BodySmall" style={{marginTop: 20}}>some details...</Text>
-      <ScrollView style={{maxHeight: 100, padding: 10}}>
-        <Text type="BodySmall" style={globalStyles.selectable}>{safeError}</Text>
-        <Text type="BodySmall" style={globalStyles.selectable}>{safeInfo}</Text>
+      <Text type="BodySmall" style={detailHeaderStyle}>Error details</Text>
+      <Text type="BodySmall" style={{...globalStyles.selectable, margin: 10}}>{`${name}: ${message}`}</Text>
+      <Text type="BodySmall" style={{marginTop: 20}}>Stack trace</Text>
+      <ScrollView style={detailContainerStyle}>
+        <Text type="BodySmall" style={detailStyle}>{stack}</Text>
+      </ScrollView>
+
+      <Text type="BodySmall" style={detailHeaderStyle}>Component stack trace</Text>
+      <ScrollView style={detailContainerStyle}>
+        <Text type="BodySmall" style={detailStyle}>{componentStack}</Text>
       </ScrollView>
     </Box>
   )
 }
 
-class ErrorBoundary extends React.PureComponent<any, {error: ?Error, info: ?ErrorInfo}> {
-  state = {
-    error: null,
-    info: null,
-  }
+type Props = {
+  children: React.Node,
+}
+
+type State = {
+  info: ?AllErrorInfo,
+}
 
-  componentWillReceiveProps(nextProps: any) {
+class ErrorBoundary extends React.PureComponent<Props, State> {
+  state = {info: null}
+
+  componentWillReceiveProps(nextProps: Props) {
     if (this.props.children !== nextProps.children) {
-      if (this.state.error) {
-        this.setState({error: null, info: null})
-      }
+      this.setState({})
```

Comment:
I don't think you want this. This means anything that gets passed to this might clear it out

Reply:
Hmm...how is the old behavior different, though? What should I do instead?

Reply:
Oh sorry. No I mean setstate with empty does nothing. No update

Reply:
Ah...it's a merge. Got it.